siliconcompiler/tools/openroad/openroad.py

```diff
@@ -158,14 +158,14 @@ def setup(chip, mode='batch'):
         # For each OpenROAD tool variable, read default from main library and write it
         # into schema. If PDK doesn't contain a default, the value must be set
         # by the user, so we add the variable keypath as a requirement.
-        if chip.valid('library', lib, 'asic', 'var', tool, variable):
-            value = chip.get('library', lib, 'asic', 'var', tool, variable)
+        if chip.valid('library', mainlib, 'asic', 'var', tool, variable):
+            value = chip.get('library', mainlib, 'asic', 'var', tool, variable)
             # Clobber needs to be False here, since a user might want to
             # overwrite these.
             chip.set('tool', tool, 'var', step, index, variable, value,
                      clobber=False)
 
-        keypath = ','.join(['library', lib, 'asic', 'var', tool, variable])
+        keypath = ','.join(['library', mainlib, 'asic', 'var', tool, variable])
         chip.add('tool', tool, 'require', step, index, keypath)
```

Comment:
I think this block might want to be indented under the `chip.valid(...)` if statement, since the requirement seems optional. I think this would still fail out if the mainlib didn't actually define a variable.

(it's still valuable to conditionally add the keypath to "require" since this will make sure SC checks that the paths resolve correctly before runtime).